berdnikovsergey-lesson2/js/main.js

```diff
@@ -0,0 +1,196 @@
+
+//1. Почему код дает именно такие результаты?
+
+var a = 1 , b = 1 , c, d;
+c = ++a; alert(c); // 2  - возврвщается значеие уже инкрементируемое при префиксктной форме
+d = b++; alert(d); // 1 -  возврвщается текущее значеие при постфикстной форме
+c = ( 2 + ++a); alert(c); // 5 - поскольку мы инкрементиковали 'a(2)' в первом выражении,
+//то второй префикстный инкремент даст нам в инкременте 'a(3)' и сумма '2 + a(3) = 5'
+d = ( 2 + b++); alert(d); // 4 - поскольку мы инкрементировали 'b++(2)' то сумма 2 + 'b++(2) = 4',
+//постфикстный инкремент в текущем выражении не участвует
+alert(a); // 3 - переменая 'a' инкрементировалась 3 раза, возвратил текущее число инкрементов при префикстной форме
+alert(b); // 3 - переменая 'b' инкрементировалась 3 раза, но возвратил на 1 инкремент меньше, постфикстная форма
+
+
+//2. Чему будет равен x ?
+
+ var a = 2 ;
+ var x = 1 + (a *= 2 ); // x = 1 + ( a = 2 * 2) = 5
+
+/*
+3. Объявить две целочисленные переменные — a и b и задать им произвольные начальные
+значения. Затем написать скрипт, который работает по следующему принципу:
+o если a и b положительные, вывести их разность;
+o если а и b отрицательные, вывести их произведение;
+o если а и b разных знаков, вывести их сумму;
+Ноль можно считать положительным числом.
+
+ */
+
+var result = function calc( a, b ) {
+       if ( (a >= 0) && (b >= 0) )
+           return a - b;
+       else if( (a < 0) && (b < 0) )
+           return a * b;
+       else
+           return a + b;
+}
+alert( result( -5, 1 ) );
+
+/*
+4. Присвоить переменной а значение в промежутке [0..15]. С помощью оператора switch
+организовать вывод чисел от a до 15.
+ */
+
+var a = prompt('Введите число: ');
+if (a > 0 && a < 15) {
+    switch (a) {
+        case '0' :
+            console.log(0);
+        case '1' :
+            console.log(1);
+        case '2' :
+            console.log(2);
+        case '3' :
+            console.log(3);
+        case '4' :
+            console.log(4);
+        case '5' :
+            console.log(5);
+        case '6' :
+            console.log(6);
+        case '7' :
+            console.log(7);
+        case '8' :
+            console.log(8);
+        case '9' :
+            console.log(9);
+        case '10' :
+            console.log(10);
+        case '11' :
+            console.log(11);
+        case '12' :
+            console.log(12);
+        case '13' :
+            console.log(13);
+        case '14' :
+            console.log(14);
+        case '15' :
+            console.log(15);
+    }
+}else console.log('Значение не в диавазоне');
```

Comment:
Рекомендую использовать фигурные скобки всегда - так читаемость лучше
```suggestion
} else {
  console.log('Значение не в диавазоне');}
```